code/model/modelreplace.cpp

```diff
@@ -218,12 +236,31 @@ int reallocate_and_copy_array(T*& array, int& size, size_t to_add) {
 
 	//Copy over old data. Pointers in the struct can still point to old members, we will just delete the outer bsp_info array
 	for (int i = 0; i < size_before; i++)
-		array[i] = oldArray[i];
+		array[i] = std::move(oldArray[i]);
 	delete[] oldArray;
 
 	return size_before;
 }
 
+template<typename T>
+int reallocate_and_copy_array_vmalloc(T*& array, int& size, size_t to_add) {
+	//Make sure to keep old data
+	T* oldArray = array;
+
+	int size_before = size;
```

Comment:
because the old code uses int's as array-size-tracking variables, and I don't want any conversion shenanigans when I bind the non-const reference of that, so the external-facing reference type is an int.